UPGRADING.rst

```diff
@@ -10,6 +10,49 @@ branch that does not have the listed changes, the steps would need to be
 reverted. This is all fairly informal and loosely defined. Hopefully we won't
 have too many entries in this file.
 
+#4958 Storage bucket is still being removed from TF state
+=========================================================
+
+Everyone
+~~~~~~~~
+
+PR #4926 left in place code to remove the S3 storage bucket from the Terraform
+state. There are three possible states to consider when upgrading a deployment.
+Rebase all local branches on ``develop`` before deploying any changes.
+
+1) If you have already deployed the changes for PR #4926, and have not attempted
+to deploy any subsequent changes, no action is needed. Verify that
+``terraform state show aws_s3_bucket.storage`` shows a valid bucket state before
+deploying any changes.
+
+2) If you have already deployed the changes for PR #4926, and *have* attempted
+to deploy subsequent changes, the affected deployment needs to be repaired.
+``terraform state show aws_s3_bucket.storage`` should fail with the message
+"No instance found for the given address!" To repair the deployment, run::
+
+    cd terraform
+    make plan
+    terraform import aws_s3_bucket.storage $AZUL_S3_BUCKET
+    make apply
+
+And confirm that ``terraform state show aws_s3_bucket.storage`` now shows a
+valid bucket state.
+
+3) If you have *not* yet deployed the changes for PR #4926, checkout and deploy
+the merge commit for that PR before deploying any subsequent changes.
+``terraform state show aws_s3_bucket.storage`` should show a valid bucket state
+before and after each commit is deployed.
+
+Operator
+~~~~~~~~
+
+Follow the instructions in case 1 for ``sandbox``, ``dev``, ``anvilbox``, and
+``anvildev``. Announce for other developers to upgrade their personal
+deployments.
+
+When promoting to ``prod``, follow the instructions in case 3.
```

Comment:
The promotion is a single merge commit that includes the combined changes from this PR and #4926. There is no easy way to checkout the intermediate commit. I think it would be better to solve case 3 by running `terraform state rm` prior to deploying the changes from this PR, without checking out any intermediate commits.